### PR TITLE
Fixup documentation.

### LINKED ISSUE #2896: documentation fixups
#### Summary:
There are some typos to be cleaned up, see the changeset in #2897.

#### Current Version:
v2.22.0

### PR DESCRIPTION
**Pull request by peterwicksstringfield**
#### Submission Checklist

- [X] Run unit tests: `./runTests.py src/test/unit`
- [X] Run cpplint: `make cpplint`
- [X] Declare copyright holder and open-source license: see below

#### Summary

Fixes some typos in documentation and comments.

Since these aren't typos in the manual, but rather in the doxygen and the comments, I made an actual PR, instead of just listing the fixes for later. I hope that this is correct.

Closes #2896.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Peter Wicks Stringfield

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/stan/io/util.hpp [hunk 1]
```diff
@@ -16,14 +16,12 @@ namespace io {
  * @return String representation of current UTC.
  */
 std::string utc_time_string() {
-  // FIXME:  use std::strftime
-
   // original with asctime
```
Comment (peterwicksstringfield) [inline src/stan/io/util.hpp]: Notice below where strftime is called. This fixme has already been fixed and can be deleted.

File: src/stan/lang/generator/generate_validate_nonnegative.hpp [hunk 2]
```diff
@@ -18,12 +18,8 @@ namespace lang {
...
  * Called by
- * <br />generate_validate_context_size - data variables
- * <br />generate_initialization - transformed data declarations
- * <br />generate_var_resiszing - initializes transformed data variables
- * <br />generate_local_var_decl_inits - local variables, transformed parameters
- *                              write array, generated quantities
- * <br/> generate_set_param_ranges - parameter variables
+ * <br />generate_validate_var_dims
+ * <br/> generate_set_param_ranges
  *
```
Comment (peterwicksstringfield) [inline src/stan/lang/generator/generate_validate_nonnegative.hpp]: This list was out of date. I updated it, but I think we should just remove it.

Comment (rok-cesnovar) [reply src/stan/lang/generator/generate_validate_nonnegative.hpp]: Thanks. This code is no longer being maintained as the stan compiler written in c++ (src/stan/lang folder) is currently only being used to debug stanc3, but I guess it doesnt hurt to clean this up.

Comment (peterwicksstringfield) [reply src/stan/lang/generator/generate_validate_nonnegative.hpp]: Thanks for explaining that. On the plus side, documentation can't go out of sync with code that isn't changing anymore :).

**Pull request merged**